pkg/controller/operators/olm/operator.go

```diff
@@ -241,6 +279,212 @@ func (a *Operator) syncRBAC(obj interface{}) (syncError error) {
 	return nil
 }
 
+func (a *Operator) updateDeploymentAnnotation(op *v1alpha2.OperatorGroup) error {
+	// NOTE: if a CSV modification is required in the future, copy the original
+	// data as done in the bottom of this method first.
+
+	selector, err := metav1.LabelSelectorAsSelector(&op.Spec.Selector)
+	if err != nil {
+		return err
+	}
+	operatorGroupOpts := metav1.ListOptions{LabelSelector: selector.String()}
+	// TODO(jpeeler): this needs to use user impersonation with op.Spec.ServiceAccount
+	namespaceList, err := a.OpClient.KubernetesInterface().CoreV1().Namespaces().List(operatorGroupOpts)
+	if err != nil {
+		return err
+	}
+
+	nsCount := len(namespaceList.Items)
+
+	if len(op.Status.Namespaces) == nsCount {
+		nsMap := map[string]struct{}{}
+		for _, v := range namespaceList.Items {
+			nsMap[v.Name] = struct{}{}
+		}
+		equalityCheck := true
+		for _, v := range op.Status.Namespaces {
+			if _, ok := nsMap[v.Name]; !ok {
+				equalityCheck = false
+				break
+			}
+		}
+		if equalityCheck {
+			// status is current with correct namespaces, so no further updates required
+			return nil
+		}
+	}
+	op.Status.Namespaces = namespaceList.Items
+	op.Status.LastUpdated = timeNow()
+
+	// make string list made up of comma separated namespaces
+	var nsList strings.Builder
+	for i := 0; i < nsCount-1; i++ {
+		nsList.WriteString(namespaceList.Items[i].Name + ",")
+	}
+	nsList.WriteString(namespaceList.Items[nsCount-1].Name)
+
+	currentNamespace := op.GetNamespace()
+	csvsInNamespace := a.csvsInNamespace(currentNamespace)
+	for csvName, csv := range csvsInNamespace {
+		strategy, err := a.resolver.UnmarshalStrategy(csv.Spec.InstallStrategy)
+		if err != nil {
+			return fmt.Errorf("error unmarshaling strategy from ClusterServiceVersion '%s' with error: %s", csvName, err)
+		}
+
+		strategyDetailsDeployment, ok := strategy.(*install.StrategyDetailsDeployment)
+		if !ok {
+			return fmt.Errorf("could not assert strategy implementation as deployment for CSV %s", csvName)
+		}
+
+		managerPolicyRules := []rbacv1.PolicyRule{}
```

Comment:
the rbac generation is broadly doing the right thing, but the installplan takes care of some of this already:

1. for any CRD it installs, it creates a 
   1. `<that-crd>-edit` ClusterRole
   1. `<that-crd>-view` ClusterRole

All that's really left for this control loop to do is make sure there's an `<operatorgroup>-edit` and an `<operatorgroup>-view` ClusterRole available (which I see that you have!)

So I think most of this will remain the same, you just don't need to worry about the specific edit/view clusterroles